Sources/SymbolKit/SymbolGraph/Symbol/Symbol.swift

```diff
@@ -416,6 +386,35 @@ extension SymbolGraph.Symbol {
             try container.encode(identifier)
         }
 
+        /// Check the given identifier string against the list of known identifiers.
+        ///
+        /// - Parameter identifier: The identifier string to check.
+        /// - Returns: The matching `KindIdentifier` case, or `nil` if there was no match.
+        private static func lookupIdentifier(identifier: String) -> KindIdentifier? {
+            switch identifier {
+            case "associatedtype" : return .associatedtype
+            case "class"          : return .class
+            case "deinit"         : return .deinit
+            case "enum"           : return .enum
+            case "enum.case"      : return .case
+            case "func"           : return .func
+            case "func.op"        : return .operator
+            case "init"           : return .`init`
+            case "method"         : return .method
+            case "property"       : return .property
+            case "protocol"       : return .protocol
+            case "struct"         : return .struct
+            case "subscript"      : return .subscript
+            case "type.method"    : return .typeMethod
+            case "type.property"  : return .typeProperty
+            case "type.subscript" : return .typeSubscript
+            case "typealias"      : return .typealias
+            case "var"            : return .var
+            case "module"         : return .module
+            default               : return nil
+            }
+        }
```

Comment:
We are maintaining the same information as the allCases which will often lead to bug.

```suggestion
        private static func lookupIdentifier(identifier: String) -> KindIdentifier? {
			allCases.first { $0.identifier == identifier }
        }
```